common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -325,6 +330,15 @@ private void processPublish(ChannelHandlerContext ctx, MqttPublishMessage mqttMs
                 handleGatewayPublishMsg(ctx, topicName, msgId, mqttMsg);
                 transportService.reportActivity(deviceSessionCtx.getSessionInfo());
             }
+        } else if (sparkPlugSessionHandler != null) {
+            try {
+                SparkplugTopic sparkplugTopic = parseTopic(topicName);
+                log.error("Publish [{}] [{}]", sparkplugTopic.isNode() ? "node" : "device", sparkplugTopic.getType());
+                handleSparkplugPublishMsg(ctx, sparkplugTopic, msgId, mqttMsg);
+                transportService.reportActivity(deviceSessionCtx.getSessionInfo());
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
This is bad practise and I remind you this all the time :-)

Reply:
Ok